desktop/src/ui.rs

```diff
@@ -90,4 +194,66 @@ impl UiBackend for DesktopUiBackend {
             .set_buttons(MessageButtons::Ok);
         dialog.show();
     }
+
+    fn display_file_open_dialog(&mut self, filters: Vec<FileFilter>) -> Option<DialogResultFuture> {
+        // Prevent opening multiple dialogs at the same time
+        if self.dialog_open {
+            return None;
+        }
+        self.dialog_open = true;
+
+        // Create the dialog future
+        Some(Box::pin(async move {
+            let mut dialog = AsyncFileDialog::new();
+
+            for filter in filters {
+                if std::env::consts::OS == "macos" && filter.mac_type.is_some() {
```

Comment:
I think `cfg!(target_os = "macos")` can be used here, so the following code will be eliminated in non-macOS builds.